### DISCUSSION TITLE
Improve error messages on Image Transform

### DISCUSSION THREAD

**Discussion by njbmartin** [supabase/supabase#21665] [Feature Requests]
I've been trying to work out why some images being uploaded by my users are not displaying at all (image url returns an error). Images are being displayed using the Image Transformer:

```tsx
supabase.storage.from('avatars').getPublicUrl('image.jpg', {
  transform: {
    width: 960,
    height: 960,
  },
})
```

**This returns the URL just fine, but doesn't acknowledge that the source image cannot be transformed.** If I go to the URL in my browser, I get this error:

```
{"statusCode":"422","error":"ImageProcessingError","message":"Invalid source image"}
```

Took me way too long to work out why this might be the case... but in the documentation, it has the following which could probably do with being called out better.

<img width="526" alt="image" src="https://github.com/supabase/supabase/assets/5699471/4b01bf3a-ad09-4897-8074-63e33a08adb2">

While I am transforming the image to just `960x960`, the uploaded source image is `7992x8006` pixels, which exceeds that stated `50MP` limit. **It is not clear that this limit refers to the source image or the transformed image.**

**Comment by hc-12** [comment]
Agreed, the error makes you go on a wild goose chase

**Comment by IPJT** [comment]
+1!
